pkg/reconcileaction/procedures.go

```diff
@@ -0,0 +1,134 @@
+package reconcileaction
```

Comment:
somehow I dont like `reconcileaction` as the package name.

Reply:
I'll rename to `reconciler`. That would give `reconciler.Action` and `reconciler.Procedure` which seem pretty good. The caveat is that controller-runtime has a `reconcile` package that gets imported already, so we'll have `reconcile.foo` and `reconciler.foo` running around.

Reply:
`Reconsile` looks good to me. Our reconsiler can be imported by a different name , an ex: `greconsiler`